src/Settings/ListComponents.jsx

```diff
@@ -0,0 +1,236 @@
+import React from "react";
+import { Text, Linking, ScrollView } from "react-native";
+import { ListItem, Avatar, Switch, Icon } from "@rneui/themed";
+import { useSelector, useDispatch } from "react-redux";
+import STRINGS from "../common/localization";
+import styles from "./styles";
+import colors from "../common/colors";
+import FontSizeComponent from "./components/fontSize";
+import FontFaceComponent from "./components/fontFace";
+import LanguageComponent from "./components/language";
+import TransliterationComponent from "./components/transliteration";
+import ThemeComponent from "./components/theme";
+import {
+  toggleAutoScroll,
+  toggleScreenAwake,
+  toggleStatusBar,
+  toggleParagraphMode,
+  toggleStatistics,
+} from "../common/actions";
+import BaniLengthComponent from "./components/baniLength";
+import LarivaarComponent from "./components/larivaar";
+import PadchedSettingsComponent from "./components/padched";
+import VishraamComponent from "./components/vishraam";
+import TranslationComponent from "./components/translation";
+
+function ListComponent({ navigation }) {
+  const { navigate } = navigation;
+  const { isAutoScroll, isScreenAwake, isStatusBar, isParagraphMode, isStatistics, isNightMode } =
+    useSelector((state) => state);
+  const dispatch = useDispatch();
+  return (
+    <ScrollView>
+      <Text
+        style={[
+          styles.displayOptionsText,
+          {
+            backgroundColor: isNightMode ? colors.NIGHT_BLACK : colors.LABEL_COLORS,
+          },
+          { color: isNightMode ? colors.WHITE_COLOR : colors.NIGHT_BLACK },
+        ]}
+      >
+        {STRINGS.display_options}
+      </Text>
+      <FontSizeComponent isNightMode={isNightMode} dispatch={dispatch} />
+      <FontFaceComponent isNightMode={isNightMode} dispatch={dispatch} />
+      <LanguageComponent isNightMode={isNightMode} dispatch={dispatch} />
+      <TransliterationComponent isNightMode={isNightMode} dispatch={dispatch} />
+      <TranslationComponent isNightMode={isNightMode} dispatch={dispatch} />
+      <ThemeComponent isNightMode={isNightMode} dispatch={dispatch} />
+      <ListItem
```

Comment:
As we are having a number of <ListItem> components with similar definition, I was thinking if we can make a helper function to keep the code DRY. Something like this
const renderListItem = (title, icon, value, onValueChange, onPress = null) => (
    <ListItem
      key={title} // Assign a unique key based on the title or an identifier
      bottomDivider
      containerStyle={[{ backgroundColor: isNightMode ? colors.NIGHT_GREY_COLOR : null }]}
      onPress={onPress}
    >
      {icon && (
        <Icon
          color={isNightMode ? colors.COMPONENT_COLOR_NIGHT_MODE : colors.COMPONENT_COLOR}
          name={icon}
          type="material"
          size={30}
        />
      )}
      <ListItem.Content>
        <ListItem.Title style={[isNightMode && { color: colors.WHITE_COLOR }]}>
          {title}
        </ListItem.Title>
      </ListItem.Content>
      {value !== undefined && (
        <Switch value={value} onValueChange={onValueChange} />
      )}
      {onPress && <ListItem.Chevron />}
    </ListItem>
  );
Although, I am not sure about it as I have not tested this function myself. Just an idea

Reply:
Han ji veer ji I thought of this. However, I thought to segregate so that the component should work independently.  Similarities between members are low. However, we can improve this once we right all the features.